backend/mlarchive/archive/migrations/0003_fix_message_msgid.py

```diff
@@ -0,0 +1,45 @@
+# Generated by Django 4.2.13 on 2024-05-16 22:18
+
+from django.db import migrations
+import hashlib
+import base64
+
+
+def get_hash(msg):
+        """Returns the message hashcode, a SHA-1 digest of the Message-ID and listname.
+        Similar to the popular Web Email Archive, mail-archive.com
+        see: https://www.mail-archive.com/faq.html#msgid
+        """
+        msgid = msg.msgid.encode('utf8')
+        listname = msg.email_list.name.encode('utf8')
+        sha = hashlib.sha1(msgid)
+        sha.update(listname)
+        b64 = base64.urlsafe_b64encode(sha.digest())
+        return b64.decode('utf8')
+
+
+def forward(apps, schema_editor):
+    Message = apps.get_model('archive', 'Message')
+
+    msgs = Message.objects.filter(msgid__startswith=' ')
+    for m in msgs:
+        msgid = m.msgid
+        fixed_msgid = msgid.lstrip().lstrip('<')
+        m.msgid = fixed_msgid
+        m.spam_score = 20240516
```

Comment:
intriguing use of the field...

Reply:
Yes. A bit of field overloading here. This field was created to hold custom spam identifiers. Upon message ingestion or during manual spam scans, specific spam identifying functions can set this field to some value which can later be used to remove marked messages. In this case I'm just using the field to mark messages touched by this migration. I can see how this might cause confusion, "What is this for?" when viewed at a much later date. I can take this out.

Reply:
On the other hand the values are user defined and the date value does add some context. I can clear the field if no issues come up.